src/Exception/ApiException.php

```diff
@@ -50,7 +50,7 @@ public function __construct(GuzzleRequestException $previous, $message = '')
 
         if ($this->previous->hasResponse()) {
             $this->response = $this->previous->getResponse();
-            $this->requestId = $this->response->getHeader('X-Contentful-Request-Id')[0];
+            $this->requestId = $this->response->getHeaderLine('X-Contentful-Request-Id');
```

Comment:
🤷‍♂️ Once we go down the path of hardening this might as well do it right

I don't feel too strongly about this. We use `getHeader()` all over the place (check the RateLimitException classes in both SDKs for others), it might be nice to just roll a helper function somewhere that takes care of this.